controllers/appgroup_controller.go

```diff
@@ -31,6 +33,7 @@ import (
 const (
 	appgroupNameKey                   = "appgroup"
 	finalizer                         = "application-group-finalizer"
+	rwffinalizer                      = "reverse-workflow-finalizer"
```

Comment:
Can we discuss again why we are using a finalizer? Why not just execute delete when we have confirmed that the workflow has completed? The main reason I am raising this is because it block all delete of the workflow CR. This is problematic because `helm uninstall` will not be able to remove the CRO.

Reply:
Other than this, nice work @davutkarabay 🚀  looks great!

Reply:
> Can we discuss again why we are using a finalizer? Why not just execute delete when we have confirmed that the workflow has completed? The main reason I am raising this is because it block all delete of the workflow CR. This is problematic because `helm uninstall` will not be able to remove the CR.

This is mostly for better UX. We would like to show the end-user both the forward workflow (that is being reversed) and the reverse workflow on the dashboard. Can we have the hook look for finalizers and remove them during uninstall? Can we think of a reason why it may or may not be a good idea?

Reply:
Why not delete both after the reverse succeeds? How is this any different than calling the delete before with a finalizer?

Reply:
Sure - no reason why we shouldnt do that. Probably easier to follow that way. Let me take a stab at it.

Reply:
Thanks, this would simplify things and reduce conflicts :)